zerver/lib/actions.py

```diff
@@ -5786,16 +5794,18 @@ def do_update_embedded_data(
     user_profile: UserProfile,
     message: Message,
     content: Optional[str],
-    rendered_content: Optional[str],
+    rendering_result: Optional[MessageRenderingResult],
 ) -> None:
     event: Dict[str, Any] = {"type": "update_message", "message_id": message.id}
     changed_messages = [message]
+    rendered_content: Optional[str] = None
 
     ums = UserMessage.objects.filter(message=message.id)
 
     if content is not None:
-        update_user_message_flags(message, ums)
-        message.content = content
+        if rendering_result is not None:
+            update_user_message_flags(rendering_result, ums)
+            rendered_content = rendering_result["rendered_content"]
```

Comment:
Should this be an `assert rendering_result is not None`?  I think it'd be a bug if it were not present given that content is not None.

Reply:
Perhaps the function signature shouldn't let `rendering_result` be optional and then no None check on `rendering_result` will be necessary here.

Reply:
Yeah, that seems correct to me.  I hadn't thought the we had `Optional[RenderingResult]` anywhere.